src/waveform/renderers/waveformmark.cpp

```diff
@@ -5,21 +5,27 @@
 
 #include "waveformmark.h"
 
-WaveformMark::WaveformMark(int hotCue)
+WaveformMark::WaveformMark( const QString& group,
+                            const QDomNode& node,
+                            const SkinContext& context,
+                            const WaveformSignalColors& signalColors,
+                            int hotCue)
     : m_iHotCue(hotCue) {
-}
-
-void WaveformMark::reset(int hotCue) {
-    WaveformMark(hotCue).swap(*this);
-}
-
-void WaveformMark::setup(const QString& group, const QDomNode& node,
-                         const SkinContext& context,
-                         const WaveformSignalColors& signalColors) {
     QString item = context.selectString(node, "Control");
     if (!item.isEmpty()) {
         m_pPointCos = std::make_unique<ControlProxy>(group, item);
     }
-
     m_properties = WaveformMarkProperties(node, context, signalColors);
 }
+
+WaveformMark::WaveformMark(int hotCue)
+    : m_iHotCue(hotCue){
+}
+
+WaveformMark::~WaveformMark(){
+    WaveformMark tmp(kDefaultHotCue);
```

Comment:
If suppose I clear the destructor then how do the necessary swap operations will be performed